var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -0,0 +1,120 @@
+
+        is_cuda = '+cuda' in self.spec
+        is_rocm = '+rocm' in self.spec
+
+        if is_cuda or is_rocm:
+            llvm_enable_projects += ';libclc'
+
+        if is_cuda:
+            llvm_targets_to_build += ';NVPTX'
+            libclc_targets_to_build = 'nvptx64--;nvptx64--nvidiacl'
+
+        if is_rocm:
+            if sycl_build_pi_rocm_platform == 'AMD':
+                llvm_targets_to_build += ';AMDGPU'
+                libclc_targets_to_build += ';amdgcn--;amdgcn--amdhsa'
+            elif sycl_build_pi_rocm_platform and not is_cuda:
+                llvm_targets_to_build += ';NVPTX'
+                libclc_targets_to_build += ';nvptx64--;nvptx64--nvidiacl'
+
+        args = [
+            self.define_from_variant('LLVM_ENABLE_ASSERTIONS', 'no-assertions'),
+            self.define('LLVM_TARGETS_TO_BUILD', llvm_targets_to_build),
+            self.define('LLVM_EXTERNAL_PROJECTS', llvm_external_projects),
+            self.define('LLVM_EXTERNAL_SYCL_SOURCE_DIR', sycl_dir),
+            self.define('LLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR', spirv_dir),
+            self.define('LLVM_EXTERNAL_XPTI_SOURCE_DIR', xpti_dir),
+            self.define('XPTI_SOURCE_DIR', xpti_dir),
+            self.define('LLVM_EXTERNAL_XPTIFW_SOURCE_DIR', xptifw_dir),
+            self.define('LLVM_EXTERNAL_LIBDEVICE_SOURCE_DIR', libdevice_dir),
+            self.define('LLVM_ENABLE_PROJECTS', llvm_enable_projects),
+            self.define('LIBCLC_TARGETS_TO_BUILD', libclc_targets_to_build),
+            self.define('SYCL_BUILD_PI_CUDA', 'ON' if is_cuda else 'OFF'),
+            self.define('SYCL_BUILD_PI_ROCM', 'ON' if is_rocm else 'OFF'),
+            self.define('SYCL_BUILD_PI_ROCM_PLATFORM', sycl_build_pi_rocm_platform),
+            self.define('LLVM_BUILD_TOOLS', True),
+            self.define_from_variant('SYCL_ENABLE_WERROR', 'no-werror'),
+            self.define('SYCL_INCLUDE_TESTS', True),
+            self.define_from_variant('LLVM_ENABLE_DOXYGEN', 'docs'),
+            self.define_from_variant('LLVM_ENABLE_SPHINX', 'docs'),
+            self.define_from_variant('BUILD_SHARED_LIBS', 'shared-libs'),
+            self.define('SYCL_ENABLE_XPTI_TRACING', 'ON'),
+            self.define_from_variant('LLVM_ENABLE_LLD', 'use-lld'),
+            self.define_from_variant('SYCL_BUILD_PI_ESIMD_CPU', 'disable-esimd-cpu'),
+        ]
+
+        return args
+
+    def setup_run_environment(self, env):
+        bin_path = self.spec.prefix.bin
+        for env_var_name, compiler in zip(['CC', 'CXX'], ['clang', 'clang++']):
+            env.set(env_var_name, os.path.join(bin_path, compiler))
+
+        env.set('FC', str(self.compiler.fc) if self.compiler.fc else '')
+        env.set('F77', str(self.compiler.f77) if self.compiler.f77 else '')
+
+        include_env_vars = ['C_INCLUDE_PATH', 'CPLUS_INCLUDE_PATH', 'INCLUDE']
+        for var in include_env_vars:
+            env.prepend_path(var, self.prefix.include)
```

Comment:
We used to set these by default for all packages. @alalazo do you remember why we got rid of this?